packages/ibm-products/src/components/Datagrid/Datagrid/DatagridVirtualBody.tsx

```diff
@@ -84,13 +84,10 @@ const DatagridVirtualBody = (datagridState: DataGridState) => {
       const headWrapEl = document?.querySelector(
         `#${tableId} .${blockClass}__head-wrap`
       );
-      if (headWrapEl) {
+      if (headWrapEl && enableSpacerColumn) {
         headWrapEl.scrollLeft = virtualBody?.scrollLeft;
-      }
-      if (enableSpacerColumn === true) {
-        const spacerColumn: HTMLDivElement | null = document.querySelector(
-          `#${tableId} .${blockClass}__head-wrap thead th:last-child`
-        );
+        const spacerColumn: HTMLDivElement | null =
+          headWrapEl.querySelector(`thead th:last-child`);
```

Comment:
I was just thinking about this, it's possible that the spacer column isn't _always_ the last `th` element, specifically if there are row actions then the row actions will be the last `th`.